gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailConfig.java

```diff
@@ -0,0 +1,114 @@
+     * context and stores them in the message context.
+     */
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        try {
+            // Reading OAuth access token and user name from the message context
+            String oauthAccessToken =
+                    GmailUtils.lookupFunctionParam(messageContext,
+                            GmailConstants.GMAIL_PARAM_OAUTH_ACCESS_TOKEN);
+            String username =
+                    GmailUtils.lookupFunctionParam(messageContext,
+                            GmailConstants.GMAIL_PARAM_USERNAME);
+
+            // Validating the user name and OAuth access token provided by the
+            // user
+            if (username == null || "".equals(username.trim()) || oauthAccessToken == null ||
+                    "".equals(oauthAccessToken.trim())) {
+
+                String errorLog = "Invalid username or access token";
+                log.error(errorLog);
+                ConnectException connectException = new ConnectException(errorLog);
+                GmailUtils.storeErrorResponseStatus(messageContext,
+                        connectException,
+                        GmailErrorCodes.GMAIL_ERROR_CODE_CONNECT_EXCEPTION);
+                handleException(connectException.getMessage(), connectException, messageContext);
+            }
+
+            // Storing OAuth user login details in the message context
+            this.storeOauthUserLogin(messageContext, username, oauthAccessToken);
+        } catch (MessagingException e) {
+            GmailUtils.storeErrorResponseStatus(messageContext,
+                    e,
+                    GmailErrorCodes.GMAIL_ERROR_CODE_MESSAGING_EXCEPTION);
+            handleException(e.getMessage(), e, messageContext);
+        } catch (Exception e) {
+            GmailUtils.storeErrorResponseStatus(messageContext, e,
+                    GmailErrorCodes.GMAIL_COMMON_EXCEPTION);
+            handleException(e.getMessage(), e, messageContext);
+        }
+    }
+
+    /**
+     * Stores user name and access token for OAuth authentication
+     *
+     * @param messageContext
+     *            message context where the user login information should be
+     *            stored
+     * @param username
+     *            user name
+     * @param oauthAccessToken
+     *            access token
+     * @throws MessagingException
+     */
+    private void storeOauthUserLogin(MessageContext messageContext, String username,
+                                     String oauthAccessToken) throws MessagingException {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        Object loginMode = axis2MessageContext.getProperty(GmailConstants.GMAIL_LOGIN_MODE);
+        if (loginMode != null &&
+                (loginMode.toString() == GmailConstants.GMAIL_OAUTH_LOGIN_MODE) &&
```

Comment:
It is not correct to use == to check the equality of string values.. please fix..